src/components/ResultsView.tsx

```diff
@@ -485,156 +499,202 @@ const ResultsView: ResultsViewComponent = ({
         })
       );
   };
+  const debounceRef = useRef(0);
   return (
     <div
       className="roamjs-query-results-view w-full relative"
       ref={containerRef}
     >
+      {isEditSearchFilter && (
+        <div
+          className="p-4 w-full"
+          style={{
+            background: "#eeeeee80",
+          }}
+        >
+          <InputGroup
+            fill={true}
+            placeholder="Search"
+            onChange={(e) => {
+              window.clearTimeout(debounceRef.current);
+              setSearchFilter(e.target.value);
+              if (preventSavingSettings) return;
+              const resultNode = getSubTree({
+                key: "results",
+                parentUid,
+              });
+              const searchFilterNode = getSubTree({
+                key: "searchFilter",
+                parentUid: resultNode.uid,
+              });
+              debounceRef.current = window.setTimeout(() => {
+                const searchFilter = getFirstChildUidByBlockUid(
+                  searchFilterNode.uid
+                );
+                if (searchFilter)
+                  updateBlock({
+                    uid: searchFilter,
+                    text: e.target.value,
+                  });
+                else
+                  createBlock({
+                    parentUid: searchFilterNode.uid,
+                    node: { text: e.target.value },
+                  });
+              }, 1000);
+            }}
+            defaultValue={settings.searchFilter}
+          />
+        </div>
+      )}
       <Export
         isOpen={isExportOpen}
         onClose={() => setIsExportOpen(false)}
         results={allResults}
         exportTypes={getExportTypes?.(allResults)}
       />
-      <span className="absolute top-2 right-2 z-10">
-        {onRefresh && (
-          <Tooltip content={"Refresh Results"}>
-            <Button icon={"refresh"} minimal onClick={onRefresh} small />
-          </Tooltip>
-        )}
-        <Popover
-          isOpen={moreMenuOpen}
-          target={<Button minimal icon={"more"} />}
-          onInteraction={(next, e) => {
-            if (!e) return;
-            const target = e.target as HTMLElement;
-            if (
-              target.classList.contains("bp3-menu-item") ||
-              target.closest(".bp3-menu-item")
-            ) {
-              return;
-            }
-            setMoreMenuOpen(next);
-          }}
-          content={
-            isEditRandom ? (
-              <div className="relative w-72 p-4">
-                <h4 className="font-bold flex justify-between items-center">
-                  Get Random
-                  <Button
-                    icon={"small-cross"}
-                    onClick={() => setIsEditRandom(false)}
-                    minimal
-                    small
+      <div className="relative">
```

Comment:
most of the changes in this file are from adding this `div` and `prettier`

the only thing added after this line is the menu item at line 789